feature: 계좌 해지 및 자동이체 처리

try3982 opened the pull request:
## #️⃣ Issue Number
- #28 
<!--- ex) #이슈번호, #이슈번호 -->

## 📝 요약(Summary)
- 계좌 해지 시 잔액이 존재하면 사용자가 지정한 출금 계좌로 자동 이체
- 계좌 상태가 ACTIVE가 아니면 입금불가 구현
- AccountManagementService 추가

## 🛠️ PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [x] 코드 리팩토링

## 개발 상세 내역

### ✅ 1. 계좌 관련 기능 개선
-  계좌 상태가 ACTIVE가 아닐 경우 입금 불가 처리
- 최대 입금 가능 금액 100만원 초과 시 예외 처리
- 계좌 해지 시 잔액이 존재하면, 사용자가 선택한 출금 계좌로 자동 이체

### ✅ 2. 구조 개선 및 리팩토링
- 거래 내역 조회 시 정렬 및 거래유형 필터링 기능 리팩토링

### ✅ 3. ENUM  ERRORCODE추가
- AMOUNT_EXCEED_DEPOSIT_LIMIT("입금 금액은 최대 100만원까지만 허용됩니다."),

try3982 committed:
feature: 계좌 해지 및 자동이체 처리

-. 계좌 상태가 ACTIVE가 아니면 입금불가 구현
-. 최대 입금한도 100만원을 초과할경우 입근 불가
-.계좌 해지 시 잔액이 존재하면 사용자가 지정한 출금 계좌로 자동 이체

refactor:
-. AccountManagementService 생성 및 계좌 해지, 복구, 만료 삭제 책임 분리
- 트랜잭션 응답 Record 객체 일관화 (TransferResponseRecord, DepositRecord 등)
- 거래 내역 기간/정렬 조회 기능 refactoring

Paransaik commented on src/main/java/com/track/fin/controller/TransactionController.java:
```diff
@@ -68,20 +79,35 @@ public TransferResponseRecord transfer(
-    @PostMapping("/transaction/withdraw")
+    @GetMapping("/transfers")
+    public List<TransferResponseRecord> getTransfers(
+            @RequestParam String accountNumber,
```
get 조회 시 query string parameter 값이 길어질 경우 dto로 분리하시는게 깔끔하고 좋아보입니다.

Paransaik commented on src/main/java/com/track/fin/service/TransactionService.java:
```diff
@@ -34,31 +32,41 @@
+            TransactionMethodType methodType
+    ) {
         return transactionRepository.save(
                 Transaction.builder()
```
다른 생성자 처럼 from 혹은 of로 빼시는게 좋아보이네요

Paransaik committed:
fix: 순환 참조(타 repository 사용) 제거

try3982 committed:
refactor: ErrorCode 추가

try3982 committed:
refactor: 리뷰 반영
-. 필터검색을 dto로 만들어서 관리하도록 변경
-.validatePendingLoanOrAutoTransfer 메서드 분리하여 단일책임 원칙 적용

try3982 committed:
refactor: 리뷰 반영
-. 필터검색을 dto로 만들어서 관리하도록 변경
-.validatePendingLoanOrAutoTransfer 메서드 분리하여 단일책임 원칙 적용

try3982 committed:
squash! refactor: 리뷰 반영 -. 필터검색을 dto로 만들어서 관리하도록 변경 -.validatePendingLoanOrAutoTransfer 메서드 분리하여 단일책임 원칙 적용

Paransaik commented on src/main/java/com/track/fin/controller/AccountController.java:
```diff
@@ -49,27 +49,22 @@ public AccountRecord restoreAccount(
-            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDate,
-            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime endDate,
-            @RequestParam(defaultValue = "true") boolean sortDesc,
+            @ModelAttribute TransferSearchRequestRecord searchRequest,
```
1. ModelAttribute로 TransferSearchRequestRecord를 받으셨는데, 사용은 안하시네요? 필터 비지니스 로직이 들어가야하지 않나요?

2. DTO를 2개 못 받으셔서 ModelAttribute로 받으신 거 같은데, TransferRequestRecord 하나의 DTO에 filter 조건도 한번에 받으시면 안되시는 건가요?

Paransaik commented on src/main/java/com/track/fin/domain/Loan.java:
```diff
@@ -71,7 +71,7 @@ public void update(User user, Account account, CreateLoanRecord dto) {
         this.account = account;
         this.balance = dto.balance();
-     //   this.loanDate = LocalDateTime.now();
+       this.loanDate = LocalDateTime.now();
```
자동정렬 안하신 것 같습니다.

Paransaik commented on src/main/java/com/track/fin/record/TransferSearchRequestRecord.java:
```diff
@@ -0,0 +1,17 @@
+        Boolean sortDesc,
+
+        String transactionType
+) {}
```
다른 record와 같이 양식 맞춰주세요~

Paransaik commented on src/main/java/com/track/fin/service/AccountService.java:
```diff
@@ -30,8 +28,10 @@ public class AccountService {
-    private final UserRepository userRepository;
+
+    public Account createAccount(Account account) {
+        return accountRepository.save(account);
```
account 만드실 때 user 넣어주셔야 하는데, 아래 메소드 사용하시는게 맞지 않을까요?

try3982 replied:
해당 메서드를 삭제하면 될까요?

Paransaik reviewed (commented):
PR전에 스쿼시 부탁드려요!!!!!
